src/App.tsx

```diff
@@ -15,48 +24,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const getPreparedGoods = (
+  goods: Goods,
+  sortField: string,
+  isReversed: boolean,
+) => {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORTED_BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.SORTED_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT_FIELD);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const changedGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    isReversed,
+  );
+
+  const clearSort = () => {
```

Comment:
You need to choose a better name for this handler:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364